manifests/010-prometheus-rules.yaml

```diff
@@ -9,14 +9,48 @@ spec:
   groups:
   - name: SamplesOperator
     rules:
+    - alert: SamplesRetriesMissingOnImagestreamImportFailing
+      expr: sum(openshift_samples_failed_imagestream_import_info) > sum(openshift_samples_retry_imagestream_import_total) - sum(openshift_samples_retry_imagestream_import_total offset 30m)
+      for: 2h
+      labels:
+        severity: warning
+      annotations:
+        message: |
+          Samples operator is detecting problems with imagestream image imports, and the periodic retries of those
+          imports are not occurring.  Contact support.  You can look at the "openshift-samples" ClusterOperator object
+          for details. Most likely there are issues with the external image registry hosting the images that need to
+          be investigated.  The list of ImageStreams that have failing imports are:
+          {{ range query "openshift_samples_retry_imagestream_import_total > 0" }}
+            {{ .Labels.name }} {{ .Value }}
+          {{ end }}
+          However, the list of ImageStreams for which samples operator is retrying imports is:
+          retrying imports:
+          {{ range query "openshift_samples_retry_imagestream_import_total > 0" }}
+             {{ .Labels.imagestreamname }} {{ .Value }}
+          {{ end }}
+    - alert: SamplesImagestreamImportFailing
```

Comment:
This looks very similar to `SamplesRetriesMissingOnImagestreamImportFailing`. Won't they fire at the same time in most cases? What are the cases when only one of them fires?

Reply:
No they do not fire at the same time.  Testing has confirmed.
- `SamplesRetriesMissingOnImagestreamImportFailing` fires when imagestream imports are failing but for whatever reason the samples operator is *NOT* attempting to retry imports
- `SamplesImagestreamImportFailing` fires when imagestream imports are failing *AND* samples operator is retrying as expected